src/fastpam1.cpp

```diff
@@ -0,0 +1,193 @@
+/**
+ * @file fastpam1.cpp
+ * @date 2021-08-03
+ *
+ * This file contains the primary C++ implementation of the FastPAM1 code.
+ *
+ */
+#include "kmedoids_algorithm.hpp"
+
+#include <carma.h>
+#include <armadillo>
+#include <unordered_map>
+#include <regex>
+
+/**
+ * \brief Runs naive FastPAM1 algorithm.
+ *
+ * Run the naive FastPAM1 algorithm to identify a dataset's medoids.
+ *
+ * @param input_data Input data to find the medoids of
+ */
+void KMedoids::fit_fastpam1(arma::mat input_data) {
+  data = input_data;
+  data = arma::trans(data);
+  arma::rowvec medoid_indices(n_medoids);
+  // runs build step
+  KMedoids::build_fastpam1(data, medoid_indices);
+  steps = 0;
+  medoid_indices_build = medoid_indices;
+  arma::rowvec assignments(data.n_cols);
+  size_t iter = 0;
+  bool medoidChange = true;
+  while (iter < max_iter && medoidChange) {
+    auto previous(medoid_indices);
+    // runs swap step as necessary
+    KMedoids::swap_fastpam1(data, medoid_indices, assignments);
+    medoidChange = arma::any(medoid_indices != previous);
+    iter++;
+  }
+  medoid_indices_final = medoid_indices;
+  labels = assignments;
```

Comment:
appears unused; rm?

Reply:
`labels` is a global variable which stores the assignment of medoids for each datapoint.
https://github.com/ThrunGroup/BanditPAM/blob/67221ff929d8005e46358a9151795b66bce66891/src/kmedoids_algorithm.cpp#L89-L97

Reply:
OK for this PR; updated #110 to fix